<h1 align='center'>Proyecto Individual 2</h1>
<h3 align='center'><i>Telecomunicaciones</i></h3>
<h4 align='center'>Cohorte:DataPT04</h4>

# `ETL`: **`Extracción, transformación y carga de datos`**
- *Buscamos en el sitio ENACOM - Ente Nacional de Comunicaciones de Argentina.*
- *Encontrando algunos archivos interesantes de internet y telefonía fija.*
##### `Primero, partimos de los archivos de trabajo:`
1. *`Internet_Accesos-por-tecnologia.csv`*
2. *`Internet_Ingresos.csv`*
3. *`Internet_Penetracion.csv`*
4. *`telefonia_fija.csv`*
##### `Segundo, creamos los siguientes datasets y son enviados al archivo dataset:`
1. *`Tecnologia.csv`*
2. *`Ingresos.csv`*
3. *`Penetracion_internet.csv`*
4. *`Telefonia_Fija.csv`*
##### `Por último, se crea Internet.csv con la fusion de los primeros 3 datasets creados y es anexado al archivo dataset`:
1. *`Internet.csv`*
2. *`Telefonia_Fija.csv`*

In [1]:
# Importamos la libreria que necesitamos para abrir los archivos
import pandas as pd

#### `Acceso a Internet por tecnología`

In [2]:
# Abrimos el dataset
df1 = pd.read_csv('../data/Internet_Accesos-por-tecnologia.csv')
df1.head(2)

Año  Trimestre  ADSL  Cablemodem  Fibra óptica  Wireless  Otros  Total  \
0  2022          4     1           6             3       573    222     11   
1  2022          3     1           6             3       557    235     11   

         Periodo  
0   Oct-Dic 2022  
1  Jul-Sept 2022

In [3]:
# Elimino las columnas no definidas y las que no coindicen
df1 = df1.drop(['Otros','Total'],axis=1)

In [4]:
# Verificamos el resultado
df1.head()

Año  Trimestre  ADSL  Cablemodem  Fibra óptica  Wireless        Periodo
0  2022          4     1           6             3       573   Oct-Dic 2022
1  2022          3     1           6             3       557  Jul-Sept 2022
2  2022          2     1           6             3       556   Abr-Jun 2022
3  2022          1     2           6             2       546   Ene-Mar 2022
4  2021          4     2           6             2       523   Oct-Dic 2021

In [5]:
# Checamos tipo de dato y registros
df1.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 36 entries, 0 to 35
Data columns (total 7 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   Año           36 non-null     int64 
 1   Trimestre     36 non-null     int64 
 2   ADSL          36 non-null     int64 
 3   Cablemodem    36 non-null     int64 
 4   Fibra óptica  36 non-null     int64 
 5   Wireless      36 non-null     int64 
 6   Periodo       36 non-null     object
dtypes: int64(6), object(1)
memory usage: 2.1+ KB


In [6]:
# Salvamos como dataset 'Tecnologia.csv'
#df1.to_csv('Tecnologia.csv',index=False)

#### Resumen: `Acceso a Internet por tecnología`
* Limpiamos el dataset reduciendo a `7 columnas y 36 registros.`
* Salvamos el dataset como `Tecnologia.csv`
* Lo fusionaremos más adelante para trabajar el `EDA`

#### `Ingresos en internet`

In [7]:
df2 = pd.read_csv('../data/Internet_Ingresos.csv')
df2.head()

Año  Trimestre  Ingresos (miles de pesos)        Periodo
0  2022          4                         72   Oct-Dic 2022
1  2022          3                         64  Jul-Sept 2022
2  2022          2                         60   Abr-Jun 2022
3  2022          1                         55   Ene-Mar 2022
4  2021          4                         45   Oct-Dic 2021

In [8]:
# Verificamos registros, columnas y tipos de dato
df2.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 36 entries, 0 to 35
Data columns (total 4 columns):
 #   Column                     Non-Null Count  Dtype 
---  ------                     --------------  ----- 
 0   Año                        36 non-null     int64 
 1   Trimestre                  36 non-null     int64 
 2   Ingresos (miles de pesos)  36 non-null     int64 
 3   Periodo                    36 non-null     object
dtypes: int64(3), object(1)
memory usage: 1.3+ KB


In [9]:
# Salvamos el dataset como 'Ingresos.csv'
#df2.to_csv('Ingresos.csv',index=False)

#### Resumen: `Ingresos en internet`
* Limpiamos el dataset reduciendo a `4 columnas y 36 registros.`
* Salvamos el dataset como `Ingresos.csv`
* Lo fusionaremos más adelante para trabajar el `EDA`

#### `Penetracion del internet`

In [10]:
# Abrimos el dataset
df3 = pd.read_csv('../data/Internet_Penetracion.csv')
df3.head()

Año  Trimestre        Provincia Accesos por cada 100 hogares  Unnamed: 4  \
0  2022          4     Buenos Aires                        78,94         NaN   
1  2022          4  Capital Federal                       122,73         NaN   
2  2022          4        Catamarca                        65,77         NaN   
3  2022          4            Chaco                        43,62         NaN   
4  2022          4           Chubut                        85,57         NaN   

   Unnamed: 5  Unnamed: 6  
0         NaN         NaN  
1         NaN         NaN  
2         NaN         NaN  
3         NaN         NaN  
4         NaN         NaN

In [11]:
# Vemos que la datos de la columna: 'Accesos por cada 100 hogares', aparentemente son 'object'
# pero ellos lucen como 'float' y separado por coma.
df3.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 864 entries, 0 to 863
Data columns (total 7 columns):
 #   Column                        Non-Null Count  Dtype  
---  ------                        --------------  -----  
 0   Año                           864 non-null    int64  
 1   Trimestre                     864 non-null    int64  
 2   Provincia                     864 non-null    object 
 3   Accesos por cada 100 hogares  864 non-null    object 
 4   Unnamed: 4                    0 non-null      float64
 5   Unnamed: 5                    0 non-null      float64
 6   Unnamed: 6                    0 non-null      float64
dtypes: float64(3), int64(2), object(2)
memory usage: 47.4+ KB


In [12]:
# Entonces, re-abrimos el dataset con el parámetro "decimal=','" para buscar una solución.
# Vemos, ahora el cambio de coma a punto decimal en la columna mencionada.
df3 = pd.read_csv('../data/Internet_Penetracion.csv',decimal=',')
df3.head()

Año  Trimestre        Provincia  Accesos por cada 100 hogares  Unnamed: 4  \
0  2022          4     Buenos Aires                         78.94         NaN   
1  2022          4  Capital Federal                        122.73         NaN   
2  2022          4        Catamarca                         65.77         NaN   
3  2022          4            Chaco                         43.62         NaN   
4  2022          4           Chubut                         85.57         NaN   

   Unnamed: 5  Unnamed: 6  
0         NaN         NaN  
1         NaN         NaN  
2         NaN         NaN  
3         NaN         NaN  
4         NaN         NaN

In [13]:
# Ahora vemos que el tipo de dato de la correspondiente columna es 'float'
df3.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 864 entries, 0 to 863
Data columns (total 7 columns):
 #   Column                        Non-Null Count  Dtype  
---  ------                        --------------  -----  
 0   Año                           864 non-null    int64  
 1   Trimestre                     864 non-null    int64  
 2   Provincia                     864 non-null    object 
 3   Accesos por cada 100 hogares  864 non-null    float64
 4   Unnamed: 4                    0 non-null      float64
 5   Unnamed: 5                    0 non-null      float64
 6   Unnamed: 6                    0 non-null      float64
dtypes: float64(4), int64(2), object(1)
memory usage: 47.4+ KB


In [14]:
# Ahora, eliminamos las columnas no definidas y con datos nulos
df3 = df3.drop(['Unnamed: 4','Unnamed: 5','Unnamed: 6'],axis=1)

In [15]:
# Verificamos el resultado
df3.head()

Año  Trimestre        Provincia  Accesos por cada 100 hogares
0  2022          4     Buenos Aires                         78.94
1  2022          4  Capital Federal                        122.73
2  2022          4        Catamarca                         65.77
3  2022          4            Chaco                         43.62
4  2022          4           Chubut                         85.57

In [16]:
# Enseguida, renombramos la columna 'Accesos por cada 100 hogares'
# como 'Accesoactual'
df3 = df3.rename(columns={'Accesos por cada 100 hogares':'Accesoactual'})

In [17]:
# Verificando el resultado
df3.head()

Año  Trimestre        Provincia  Accesoactual
0  2022          4     Buenos Aires         78.94
1  2022          4  Capital Federal        122.73
2  2022          4        Catamarca         65.77
3  2022          4            Chaco         43.62
4  2022          4           Chubut         85.57

In [18]:
# Checamos registros, columnas y tipo de dato
df3.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 864 entries, 0 to 863
Data columns (total 4 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   Año           864 non-null    int64  
 1   Trimestre     864 non-null    int64  
 2   Provincia     864 non-null    object 
 3   Accesoactual  864 non-null    float64
dtypes: float64(1), int64(2), object(1)
memory usage: 27.1+ KB


In [19]:
# Salvamos el dataset como
#df3.to_csv('Penetracion_internet.csv',index=False)

#### Resumen: `Penetracion del internet`
* Limpiamos el dataset reduciendo a `4 columnas y 864 registros.`
* Salvamos el dataset como `Penetracion_internet.csv`
* Lo reservamos para usar en `PowerBI`
* Lo fusionaremos más adelante para trabajar el `EDA`

#### `Telefonía fija`

In [20]:
df4 = pd.read_csv('../data/telefonia_fija.csv')
df4.head()

Año  Trimestre        Provincia  Accesos telefonía fija por cada 100 hab  \
0  2014          1     Buenos Aires                                    2.672   
1  2014          1  Capital Federal                                    5.266   
2  2014          1        Catamarca                                    1.003   
3  2014          1            Chaco                                  909.000   
4  2014          1           Chubut                                    1.766   

   Accesos telefonía fija por cada 100 hogares  
0                                        8.526  
1                                       12.857  
2                                        3.945  
3                                        3.402  
4                                        5.759

In [21]:
# Eliminamos la columna 'Accesos telefonía fija por cada 100 hab'
df4 = df4.drop(['Accesos telefonía fija por cada 100 hab'],axis=1)

In [22]:
# Verificamos el resultado
df4.head()

Año  Trimestre        Provincia  \
0  2014          1     Buenos Aires   
1  2014          1  Capital Federal   
2  2014          1        Catamarca   
3  2014          1            Chaco   
4  2014          1           Chubut   

   Accesos telefonía fija por cada 100 hogares  
0                                        8.526  
1                                       12.857  
2                                        3.945  
3                                        3.402  
4                                        5.759

In [23]:
# Enseguida, renombramos la columna 'Accesos por cada 100 hogares'
# como 'Accesoactual'
df4 = df4.rename(columns={'Accesos telefonía fija por cada 100 hogares':'AccesoactualTel'})

In [24]:
# Checamos resultados
df4.head()

Año  Trimestre        Provincia  AccesoactualTel
0  2014          1     Buenos Aires            8.526
1  2014          1  Capital Federal           12.857
2  2014          1        Catamarca            3.945
3  2014          1            Chaco            3.402
4  2014          1           Chubut            5.759

In [25]:
# Verificamos registros, columnas y tipo de datos.
df4.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 864 entries, 0 to 863
Data columns (total 4 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   Año              864 non-null    int64  
 1   Trimestre        864 non-null    int64  
 2   Provincia        864 non-null    object 
 3   AccesoactualTel  864 non-null    float64
dtypes: float64(1), int64(2), object(1)
memory usage: 27.1+ KB


In [26]:
# Salvamos el dataset como 'Telefonia_Fija.csv'
#df4.to_csv('Telefonia_Fija.csv',index=False)

#### Resumen: `Telefonía fija`
* Limpiamos el dataset reduciendo a `4 columnas y 864 registros.`
* Salvamos el dataset como `Telefonia_Fija.csv`
* Lo reservamos para usar en `PowerBI`
* Lo fusionaremos más adelante para trabajar el `EDA`

## `Abrimos los archivos creados`

In [27]:
df1 = pd.read_csv('./dataset/Ingresos.csv')
df2 = pd.read_csv('./dataset/Tecnologia.csv')
df3 = pd.read_csv('./dataset/Penetracion_internet.csv')
df4 = pd.read_csv('./dataset/Telefonia_Fija.csv')

### *`Obtenemos el dataset df`*: Producto de la fusión df1 y df2 

In [28]:
# Abrimos el dataset de ingresos del internet
df1.head(2)

Año  Trimestre  Ingresos (miles de pesos)        Periodo
0  2022          4                         72   Oct-Dic 2022
1  2022          3                         64  Jul-Sept 2022

In [29]:
df1.shape

(36, 4)

In [30]:
# Abrimos el dataset de tecnología del internet
df2.head(2)

Año  Trimestre  ADSL  Cablemodem  Fibra óptica  Wireless        Periodo
0  2022          4     1           6             3       573   Oct-Dic 2022
1  2022          3     1           6             3       557  Jul-Sept 2022

In [31]:
df2.shape

(36, 7)

In [32]:
# Fusionamos los archivos anteriores
df = pd.merge(df2,df1, on=['Año','Trimestre','Periodo'])

In [33]:
# Verificamos el resultado
df.head(2)

Año  Trimestre  ADSL  Cablemodem  Fibra óptica  Wireless        Periodo  \
0  2022          4     1           6             3       573   Oct-Dic 2022   
1  2022          3     1           6             3       557  Jul-Sept 2022   

   Ingresos (miles de pesos)  
0                         72  
1                         64

### *`Obtenemos el dataset df_`*: Producto de la fusión df3 y df 

In [34]:
# Abrimos el dataset de penetracion del internet
df3.head(2)

Año  Trimestre        Provincia  Accesoactual
0  2022          4     Buenos Aires         78.94
1  2022          4  Capital Federal        122.73

In [35]:
df3.shape

(864, 4)

In [36]:
# Fusionamos los archivos anteriores
df_ = pd.merge(df3,df,on=['Año','Trimestre'])

In [37]:
# Verificamos el resultado
df_.head(2)

Año  Trimestre        Provincia  Accesoactual  ADSL  Cablemodem  \
0  2022          4     Buenos Aires         78.94     1           6   
1  2022          4  Capital Federal        122.73     1           6   

   Fibra óptica  Wireless       Periodo  Ingresos (miles de pesos)  
0             3       573  Oct-Dic 2022                         72  
1             3       573  Oct-Dic 2022                         72

In [38]:
df_.shape

(864, 10)

In [39]:
# Salvamos como 'Internet.csv'
#df_.to_csv('Internet.csv',index=False)

#### *`Obtenemos el dataset Internet.csv, desde`*:
* `Ingresos.csv`
* `Tecnologia.csv`
* `Penetracion_internet.csv`

#### *`Los siguientes archivos los reservamos para el EDA`*:
- **`Internet.csv`**
- **`Telefonia_Fija.csv`**

#### *`Los siguientes archivos los reservamos para los KPI's`*:
- **`Penetracion_internet.csv`**
- **`Telefonia_Fija.csv`**


<h4 align='right'><i>Pag. 1</i></h4>